op-program/client/l2/engineapi/block_processor.go

```diff
@@ -152,7 +152,15 @@ func (b *BlockProcessor) Assemble() (*types.Block, types.Receipts, error) {
 	body := types.Body{
 		Transactions: b.transactions,
 	}
+	if b.dataProvider.Config().IsPrague(b.header.Number, b.header.Time) {
+		_requests := [][]byte{}
+		// EIP-6110 - no-op because we just ignore all deposit requests, so no need to parse logs
+		// EIP-7002
+		core.ProcessWithdrawalQueue(&_requests, b.evm)
+		// EIP-7251
+		core.ProcessConsolidationQueue(&_requests, b.evm)
 
```

Comment:
Are there other side effects of these functions? They look to be a noop

Reply:
I think this call removes requests from the queue if the contract exists at the address.

Reply:
The side effect is extra state accesses for the system contracts. Which is OK as it's expected from op-geth (and implied by the spec) so there will be a witness available for the access.

Reply:
Also, the state root does change due to these calls - if any requests are sent and we don't process these calls, we'll get a mismatched state root. https://eips.ethereum.org/EIPS/eip-7251#system-call